docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,180 @@
+
+There is also a risk that a Nextcloud application password used for something else is compromised and is used to access the user's mailbox before it is revoked. Discussion of that risk falls outside of the scope of this scenario.
+
+To answer the questions asked earlier for this specific scenario:
+
+1. Do I want to use Lua to identify mailboxes and verify that users are are authorized to use mail services? **No. Provisioning is done through LDAP.**
+1. Do I want to use Lua to verify passwords that users authenticate with for IMAP/POP3/SMTP in their mail clients? **Yes. Password authentication is done through Lua.**
+1. If the answer is 'yes' to question 1 or 2: are there other methods that better facilitate my use case instead of custom scripts which rely on me being a developer and not just a user? **No. Only HTTP can be used to authenticate against Nextcloud, which is not supported by Dovecot or DMS.**
+
+While it is possible to extend what Nextcloud supports with [Nextcloud apps](https://apps.nextcloud.com/), there is currently a mismatch between what DMS supports and what Nextcloud applications support. This might change in the future. For now, Lua will be used to bridge the gap between DMS and Nextcloud for authentication only (Dovecot passdb), while LDAP will still be used to identify mailboxes and verify authorization (Dovecot userdb).
+
+## Modify Dovecot's configuration
+
+Add the following volume values to the relevant part of `compose.yaml`:
+
+???+ example
+
+    ```yaml
+        # All new volumes are marked :ro to configure them as read-only, since their contents are not changed from inside the container
+        volumes:
+          # Configuration override to disable LDAP authentication
+          - ./docker-data/dms/config/dovecot/auth-ldap.conf.ext:/etc/dovecot/conf.d/auth-ldap.conf.ext:ro
+          # Configuration addition to enable Lua authentication
+          - ./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf:/etc/dovecot/conf.d/auth-lua-httpbasic.conf:ro
+          # Directory containing Lua scripts
+          - ./docker-data/dms/config/dovecot/lua/:/etc/dovecot/lua/:ro
+    ```
+
+Make the necessary changes on the filesystem:
+```bash
+mkdir -p ./docker-data/dms/config/dovecot/lua
+docker cp CONTAINER_NAME:/etc/dovecot/conf.d/auth-ldap.conf.ext ./docker-data/dms/config/dovecot/auth-ldap.conf.ext
+```
```

Comment:
Regarding the prior review feedback with your response: https://github.com/docker-mailserver/docker-mailserver/pull/3579#discussion_r1366689491

> The reason why I let the user make the directory beforehand is so that DMS does not have to be restarted (which would be overkill just to create a single directory).
> These files are read-only anyway. I agree that handholding is not necessary.

We have different perspectives. I assume the user does not already have a container running, especially since the earlier documentation requires configuration change to `compose.yaml` for volumes.

The lua directory will be created by Docker during that startup if it doesn't exist (_albeit with ownership of the Docker daemon user/group, which is sometimes inconvenient_).

If the user is providing Lua files here as you'd expect...they'd be capable of creating the directory themselves? So I questioned why `mkdir` was necessary to mention at all.

---

> As noted earlier, it is assumed that DMS is already up-and-running with LDAP authentication.
> Just referring to docker cp without any further hand-holding is fine.

Either I'm tired and missing the mention of a running DMS container with LDAP, or the guide to this point only mentions DMS will be configured to use LDAP, not that one is already setup and running.

> If a user does not realize a container needs to be running to copy files from it, the user would better not be messing around with Lua in the first place.

I'm not concerned about the user awareness with the `docker cp` command, but if it has any real relevance at all. It seems it can be dropped, as all you're doing is making a copy and editing it, you could just as easily create the file since it's being mounted back into the same place within the container. The user is reading a guide, copy/paste is fine.

---

> I disagree.
> The first line creates the lua directory so the user can work with it immediately.
> The second line copies the existing LDAP configuration file from the current container. This is better compared to including static configuration in the guide

The lua dir has been clarified above, I really don't see why `mkdir` is necessary but I'm not bothered leaving it there.

However since `docker cp` doesn't seem necessary, it really seems like this step is redundant.

You argue that it's better to copy than provide a static config, while the related [review feedback discussion](https://github.com/docker-mailserver/docker-mailserver/pull/3579#discussion_r1357902407) regarding this notes that `docker cp` won't be resilient to such:
- In an already existing setup after this guide, no change will be handled by the user (_unless it's documented in release notes and they read that when upgrading_), thus broken setup. This has happened with existing configs where users have replaced them entirely only to change one line.
- If such a change happens, new users following the guide may find the `docker cp` instruction is either copying over a config no longer used/enabled for DMS LDAP support, or with changes since this guide was last updated that may be incompatible (_nothing comes to mind, other than some runtime modification during startup_).

My main concern is that it's rather implicit on the expectation of the contents. The main goal is to disable the `passdb` config, but the docs could better communicate/document the expectation of what `userdb` config was at the time of the guide being last updated.

Not a major concern, since one can use `git blame` on the docs, and check what version of DMS it was written for, then pull that image to get the contents. If it's going to be fragile regardless, being more explicit seems better 🤷‍♂️ 

```
userdb {
  driver = ldap
  args = /etc/dovecot/dovecot-ldap.conf.ext
}
```

Reply:
`docker cp` has been removed. Instead, the user creates a new file with the stated example `userdb` content. Change will be visible in the next commit.